examples/stable_diffusion_v2/tools/safety_checker/safety_checker_1.py

```diff
@@ -0,0 +1,224 @@
+            import numpy as np
+            from PIL import Image
+
+            images_ = []
+            for i in range(images.shape[0]):
+                im = Image.fromarray((255.0 * images[i].transpose((1, 2, 0))).astype(np.uint8).asnumpy())
+                im = im.resize((self.image_size, self.image_size))
+                im = ms.Tensor(np.asarray(im), self.dtype)
+                images_.append(im)
+            images = ops.stack(images_).transpose((0, 3, 1, 2))
+
+        image_features = self.model.get_image_features(images)
+
+        nsfw_sim = self.cosine_distance(image_features, self.nsfw_features)
+        special_sim = self.cosine_distance(image_features, self.special_features)
+
+        scores = self.eval_safety(special_sim, nsfw_sim)
+
+        has_nsfw_concepts = [len(res["bad_concepts"]) > 0 for res in scores]
+
+        if self.backend == "pt":
+            import torch
+        for idx, has_nsfw_concept in enumerate(has_nsfw_concepts):
+            if has_nsfw_concept:
+                if self.backend == "pt":
+                    original_images[idx] = torch.zeros_like(original_images[idx])
+                else:
+                    original_images[idx] = ops.zeros(original_images[idx].shape)
+
+        if any(has_nsfw_concepts):
+            print(
+                "Potential NSFW content was detected in one or more images. A black image will be returned instead."
+                " Try again with a different prompt and/or seed."
+            )
+
+        return original_images, has_nsfw_concepts
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--config",
+        default="tools/_common/clip/configs/clip_vit_l_14.yaml",
+        type=str,
+        help="YAML config files for ms backend" " Default: tools/_common/clip/configs/clip_vit_l_14.yaml",
+    )
+    parser.add_argument(
+        "--model_name",
+        default="openai/clip-vit-large-patch14",
+        type=str,
+        help="the name of a (Open/)CLIP model as shown in HuggingFace for pt backend."
+        " Default: openai/clip-vit-large-patch14",
+    )
+    parser.add_argument(
+        "--image_path_or_dir",
+        default=None,
+        type=str,
+        help="input data for predict, it support real data path or data directory." " Default: None",
+    )
+    parser.add_argument("--ckpt_path", default=None, type=str, help="load model checkpoint." " Default: None")
+    parser.add_argument(
+        "--backend",
+        default="ms",
+        type=str,
+        help="backend to do CLIP model inference for CLIP score compute. Option: ms, pt." " Default: ms",
+    )
+    parser.add_argument(
+        "--tokenizer_path",
+        default="ldm/models/clip/bpe_simple_vocab_16e6.txt.gz",
+        type=str,
+        help="load tokenizer checkpoint." " Default: ldm/models/clip/bpe_simple_vocab_16e6.txt.gz",
+    )
+    parser.add_argument(
+        "--settings_path",
```

Comment:
For simplicity, let us just use one argument to parse config file. In `safety_setting_1.yaml` (please rename to `safety_checker_v1.yaml`),  indicate the path to the config file of clip model.

Reply:
Both `safety_setting_1.yaml` and `safety_setting_2.yaml` (they contain nsfw terms) are for safety checker v1. If we do this, we still need `--config` arg for v2. And I think it's good to have safety settings and model settings decoupled.

So maybe we can leave these two arguments unchanged?